File: 1512_number_of_good_pairs/solution.cpp
// LeetCode 1512 - Number of Good Pairs
// https://leetcode.com/problems/number-of-good-pairs/

#include <unordered_map>
#include <vector>

class Solution {
public:
    int numIdenticalPairs(std::vector<int>& nums) {
        std::unordered_map<int, int> counts;
        for (int num : nums) {
            ++counts[num];
        }
        int answer = 0;
        for (const auto& [_, count] : counts) {
            answer += count * (count - 1) / 2;
        }
        return answer;
    }
};

File: 1512_number_of_good_pairs/Solution.swift
// LeetCode 1512 - Number of Good Pairs
// https://leetcode.com/problems/number-of-good-pairs/

class Solution {
    func numIdenticalPairs(_ nums: [Int]) -> Int {
        var counts = [Int: Int]()
        for num in nums { counts[num, default: 0] += 1 }
        return counts.values.reduce(0) { $0 + $1 * ($1 - 1) / 2 }
    }
}
